### DISCUSSION TITLE
Filament import action

### DISCUSSION THREAD

**Discussion by w0r00d** [filamentphp/filament#14761] [Help] [answered]
### Package

Actions

### Package Version

v3.0.0

### How can we help you?

I am using import action to upload data to the database. I want one of the columns to have a value from a select list and not uploaded from the file. 

Can this be implemented?

**Comment by leandrocfe** [comment]
Maybe [Using import options](https://filamentphp.com/docs/3.x/actions/prebuilt-actions/import#using-import-options) ?

**Reply by w0r00d** [reply]
As far as I understand 'the import options' allow to alter the behavior of the importer overall and not add specific values or details. 

I'd be happy if i am missing something that can resolve that.

**Reply by leandrocfe** [reply, marked as answer]
You can do something like this:

```php
public static function getOptionsFormComponents(): array
{
    return [
        Select::make('custom')
        ->options([...]),
    ];
}

public function resolveRecord(): ?YourModel
{
    return YourModel::firstOrNew(
        [
            'field1' => $this->data['field1'],
            'field2' => $this->data['field2'],
            'your_custom_column' => $this->options['custom'],
        ]);

    return new YourModel;
}
```

**Reply by w0r00d** [reply]
Thank you very much. This works perfectly!
